### ISSUE TITLE
A User-friendly Firmware Updater Tool is required

### ISSUE THREAD

**Issue by seclorum** [magicshifter/MS3000#18]
We need an easy-to-use tool for end-users that allows them to take Firmware update products (.bin files) and flash them to their shifter's using a USB cable.

There are already a number of these tools out there - my personal favourite is BitBumper: 

http://bitbumper.de/

We should test these tools and come to a final conclusion, to be documented in an MS3000 End-User FAQ, as to how users can easily do upgrades in the future.  Since MS3KOS2.0 has OTA features this problem will hopefully dissipate over the next few months - but until then we should be armed with the details for how end-users can easily install updates going forward.

**Comment by seclorum** [comment]
http://1st.bitbumper.de/ota-firmware-update-tool-for-esp8266/

Windows-only.  TODO: find and evaluate MacOS and Linux methods.

**Comment by s00500** [comment]
https://github.com/Rodmg/esptool-gui

**Comment by seclorum** [comment]
Yeah, esptool-gui: also good considering we're already using esptool (well, a fork of it).  Needs to be much more friendly for users to install and use however: see requirements for running on MacOS for example.

**Comment by s00500** [comment]
We could fork it, it has not received new commits lately anyway...

**Comment by seclorum** [comment]
Better imho would be to find the tools that are already out there and which do the job, such as bitbumper and so on.  Unless you’re volunteering to fork and maintain the official WaW esp updater for windows, Mac and Linux? 

Btw, with OTA updates implemented this problem will become less urgent - but getting existing users flashed to the new firmware with OTA is the first purpose here... if there are existing tools for that let’s use them, because then OTA will be the principle means of updates going forward.
